src/lib/support/TestSceneTable.h

```diff
@@ -0,0 +1,528 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/scenes/SceneTableImpl.h>
+
+namespace chip {
+
+namespace SceneTesting {
+
+using FabricIndex        = chip::FabricIndex;
+using SceneTableEntry    = chip::scenes::DefaultSceneTableImpl::SceneTableEntry;
+using SceneTable         = chip::scenes::SceneTable;
+using SceneTableImpl     = chip::scenes::DefaultSceneTableImpl;
+using SceneStorageId     = chip::scenes::DefaultSceneTableImpl::SceneStorageId;
+using SceneData          = chip::scenes::DefaultSceneTableImpl::SceneData;
+using ExtensionFieldsSet = chip::scenes::ExtensionFieldsSet;
+using TransitionTimeMs   = chip::scenes::TransitionTimeMs;
+
+#define ON_OFF_CID 0x0006
+#define LV_CTR_CID 0x0008
+#define CC_CTR_CID 0x0300
+#define TEST_ENDPOINT1 0x0001
+#define TEST_ENDPOINT2 0x0099
+
+// ON OFF ATTRIBUTE IDs
+#define ON_OFF_ID 0x0000
+
+// LEVEL CONTROL ATTRIBUTE IDs
+#define CURRENT_LVL_ID 0x0000
+#define CURRENT_FRQ_ID 0x0004
+
+// COLOR CONTROL ATTRIBUTE IDs
+#define CURRENT_SAT_ID 0x0001
+#define CURRENT_X_ID 0x0003
+#define CURRENT_Y_ID 0x0004
+#define COLOR_TEMP_MIR_ID 00007
+#define EN_CURRENT_HUE_ID 0x4000
+#define C_LOOP_ACTIVE_ID 0x4002
```

Comment:
All of this is 100% in TestSceneTable.cpp, which already has many comments. Why is this duplicated here? What uses it? This file should have extension `.ipp` --> includable C++ code, since it's not a header, but rather common code. Suggest removing this file entirely.

Reply:
This file was used for on device tests. Like lib/support/TestGroupData.h. While not essential, it allowed for persistance testing and verifying the scene table didn't break anything when the device was running. It will be removed.

Reply:
File removed in latest commit.